doc/src/extras.rst

```diff
@@ -141,6 +141,262 @@ Logging cursor
+        :param apply_lsn: a LSN position up to which the warm standby server
+                          has applied the changes (physical replication
+                          master-slave protocol only)
+        :param reply: request the server to send back a keepalive message immediately
+
+        Use this method to report to the server that all messages up to a
+        certain LSN position have been stored and may be discarded.
+
+        This method can also be called with all default parameters' values to
+        send a keepalive message to the server.
+
+        In case of asynchronous connection, if the feedback message cannot be
+        sent at the moment, remembers the passed LSN positions for a later
+        hopefully successful call or call to `~flush_replication_feedback()`.
+
+    .. method:: flush_replication_feedback(reply=False)
+
+        :param reply: request the server to send back a keepalive message immediately
+
+        This method tries to flush the latest replication feedback message
+        that `~send_replication_feedback()` was trying to send, if any.
+
+    Low-level methods for asynchronous connection operation.
+
+    With the non-asynchronous connection, a single call to
+    `~start_replication()` handles all the complexity, but at times it might
+    be beneficial to use low-level interface for better control, in particular
+    to `~select.select()` on multiple sockets.  The following methods are
+    provided for asynchronous operation:
+
+    .. method:: read_replication_message(decode=True)
+
+        :param decode: a flag indicating that unicode conversion should be
+                       performed on the data received from the server
+
+        This method should be used in a loop with asynchronous connections
+        after calling `~start_replication()` once.
+
+        It tries to read the next message from the server, without blocking
+        and returns an instance of `~ReplicationMessage` or *None*, in case
+        there are no more data messages from the server at the moment.
+
+        It is expected that the calling code will call this method repeatedly
+        in order to consume all of the messages that might have been buffered,
+        until *None* is returned.  After receiving a *None* value from this
+        method, one might use `~select.select()` or `~select.poll()` on the
+        corresponding connection to block the process until there is more data
+        from the server.
+
+        The server can send keepalive messages to the client periodically.
+        Such messages are silently consumed by this method and are never
+        reported to the caller.
+
+    .. method:: fileno()
+
+        Call the corresponding connection's `~connection.fileno()` method and
+        return the result.
+
+        This is a convenience method which allows replication cursor to be
+        used directly in `~select.select()` or `~select.poll()` calls.
```

Comment:
Well, I'm not so sure about that.  The python's `select` doc says that on Windows it only works with sockets, but this is exactly our case: https://docs.python.org/2/library/select.html